.pipelines/OneBranch.PullRequest.yml

```diff
@@ -0,0 +1,161 @@
+      policheck:
+        break: true # always break the build on policheck issues. You can disable it by setting to 'false'
+      # suppression:
+      #   suppressionFile: $(Build.SourcesDirectory)\.gdn\global.gdnsuppress
+
+    stages:
+    - stage: pre_release
+      displayName: Pre-Publish
+      jobs:
+        - job: filter
+          pool:
+            type: linux
+          variables:
+            repoName: 'msal-javascript-internal'
+            ob_outputDirectory: '$(REPOROOT)/' # this directory is uploaded to pipeline artifacts, reddog and cloudvault. More info at https://aka.ms/obpipelines/artifacts
+            ob_sdl_binskim_break: true # https://aka.ms/obpipelines/sdl
+          steps:
+          - task: ShellScript@2
+            inputs:
+              scriptPath: "release-scripts/prepublish.sh"
+            name: filter
+    # PUBLISH v1
+    - stage: publish_msal_v1
+      dependsOn: pre_release
+      displayName: 'Publish MSAL v1'
+      variables:
+        publishMsalCore: $[ stageDependencies.pre_release.filter.outputs['filter.publishMsalCore'] ]
+      jobs:
+      # Release MSAL Core
+        - template: .pipelines/custom-templates/publish-template.yml@self
+          parameters:
+            jobName: 'publish_msal_core'
+            path: 'lib'
+            libName: 'msal-core'
+            cdn: true
+            publishFlagName: 'MsalCore'
+
+    # PUBLISH Common
+    - stage: publish_msal_common
+      dependsOn: pre_release
+      displayName: 'Publish MSAL Common'
+      variables:
+        publishMsalCommon: $[ stageDependencies.pre_release.filter.outputs['filter.publishMsalCommon'] ]
+      jobs:
+      # Release MSAL Common
+        - template: .pipelines/custom-templates/publish-template.yml@self
+          parameters:
+            jobName: 'publish_msal_common'
+            path: 'lib'
+            libName: 'msal-common'
+            cdn: true
+            publishFlagName: 'MsalCommon'
+
+    # PUBLISH BROWSER, REACT AND ANGULAR
+    - stage: publish_browser_packages
+      dependsOn:
+        - pre_release 
+        - publish_msal_common
+      displayName: 'Publish Browser Packages'
+      variables:
+        publishMsalBrowser: $[ stageDependencies.pre_release.filter.outputs['filter.publishMsalBrowser'] ]
+        publishMsalReact: $[ stageDependencies.pre_release.filter.outputs['filter.publishMsalReact'] ]
+        publishMsalAngular: $[ stageDependencies.pre_release.filter.outputs['filter.publishMsalAngular'] ]
+      jobs:
+        # Release MSAL Browser
+        - template: .pipelines/custom-templates/publish-template.yml@self
+          parameters:
+            jobName: 'publish_msal_browser'
+            path: 'lib'
+            libName: 'msal-browser'
+            cdn: true
+            publishFlagName: 'MsalBrowser'
+      # Release MSAL React
+        - template: .pipelines/custom-templates/publish-template.yml@self
```

Comment:
react and angular depend on browser - should these be separate stages?

Reply:
I've marked dependencies and added a success condition, keeping all the publications on the same stage since it doesn't make much difference and we'll have less artifacts this way.

Reply:
I've marked dependencies and added a success condition, keeping all the publications on the same stage since it doesn't make much difference and we'll have less artifacts this way.